### ISSUE TITLE
WebViewBridge bug??

### ISSUE THREAD

**Issue by camellieeee** [alinz/react-native-webview-bridge#154]
![2016-09-21 2 37 41](https://cloud.githubusercontent.com/assets/18331008/18700295/fae766b4-8008-11e6-9ba7-627aa0fb8b1b.png)
My HTML:  
imgs[i].onclick=function(){
 alert(1);
WebViewBridge.send('a');
alert(2);
}
only alert 1,cannot alert 2;
I think WebViewBridge is undefind. This error appears after update  XCode8

**Comment by StevenOspina** [comment]
Just noticed this too. I only get this on iOS 8 however. 9 and 10 work fine for me.

**Comment by camellieeee** [comment]
@StevenOspina Me too, but how to save it on IOS8 ?

**Comment by hmedney** [comment]
Try installing from `master` and perform a clean build - just added a small change that should improve operation under IOS 8.

**Comment by moaxaca** [comment]
@StevenOspina Also having this problem on latest master, Android 23.

**Comment by alinz** [comment]
@camellieeee did the rebuilding fix the issue? if not reopen this ticket. and give use more details about the problem.

**Comment by annalamws** [comment]
I have installed react-native-webview-bridge version 0.33.0, react-native verson 0.33.0, and I am using Xcode 8 to build to an iphone with iOS 8. Crash occurs as below:
![screen shot 2016-11-08 at 5 48 47 pm](https://cloud.githubusercontent.com/assets/23333475/20094313/ac10990e-a5db-11e6-9278-67874c6ef5c5.png)

However, iOS 9 and 10 work fine.
For iOS 8, everything work fine again when I change WebViewBridge to WebView. Please help.

**Comment by agumack** [comment]
@annalamws i have same problem

**Comment by abc1992com** [comment]
In iOS8, this problem still exists

**Comment by ezyone123** [comment]
I am having this issue:

XCode Version 8.1 (8B62)
iPhone 5s running iOS 8.1 (12B41)

Exact issue outlined above on line 
`[webView stringByEvaluatingJavaScriptFromString:webViewBridgeScriptContent];`

`webViewBridgeScriptConent = (function (window) { 'use strict'; if (window.WebViewBridge) { return; } var RNWBSchema = 'wvb'; var sendQueue = []; var receiveQueue = []; var doc = window.document; var customEvent = doc.createEvent('Event'); function callFunc(func, message) { if ('function' === typeof func) { func(message); } } function signalNative() { window.location = RNWBSchema + '://message' + new Date().getTime(); } var WebViewBridge = { __push__: function (message) { receiveQueue.push(message); setTimeout(function () { var message = receiveQueue.pop(); callFunc(WebViewBridge.onMessage, message); }, 15); }, __fetch__: function () { var messages = JSON.stringify(sendQueue); sendQueue = []; return messages; }, send: function (message) { if ('string' !== typeof message) { callFunc(WebViewBridge.onError, "message is type '" + typeof message + "', and it needs to be string"); return; } sendQueue.push(message); signalNative(); }, onMessage: null, onError: null }; window.WebViewBridge = WebViewBridge; customEvent.initEvent('WebViewBridge', true, true); doc.dispatchEvent(customEvent); }(window));
`
Issue only happens on iOS 8, other devices work just fine

**Comment by ezyone123** [comment]
So it seems like this functionality has made it into react 0.37 in the web view, this library is no longer needed and seems to have lost the support of the author. This is great as this library remade a lot of stuff that was already in the web view so now its all in one place.

[Details on the project that added this functionality](https://github.com/facebook/react-native/pull/9762#issuecomment-245635042)

[The docs](https://facebook.github.io/react-native/docs/webview.html#onmessage)

Basically replace WebViewBridge with WebView (imported from react-native) and in the web view call window.postMessage('some string'); where you want to pass a value around. The onBridgeMessage should also be changed to onMessage as a property of web view.onMessage gets an event so you must extract the value from it using `event.nativeEvent.data`
